common/djangoapps/edxmako/tests.py

```diff
@@ -41,6 +41,20 @@ def test_marketing_link(self):
             link = marketing_link('ABOUT')
             self.assertEquals(link, expected_link)
 
+        # test both variants of the 'courses' expected links
```

Comment:
At first, I was going to recommend that we actually split this out into 2 separate test methods.
But looking at it again, I think the cleanest implementation would be a single, separate method, customized with DDT.

```python
import ddt

@patch.dict('django.conf.settings.FEATURES', {'ENABLE_MKTG_SITE': False})
@ddt.data(('courses', reverse('courses')), ('https://example.com', 'https://example.com'))
@ddt.unpack
def test_marketing_link_external_courses_url(self, link_map_override, expected_link):
    with patch.dict('django.conf.settings.MKTG_URL_LINK_MAP', {'COURSES': link_map_override}):
        link = marketing_link('COURSES')
        self.assertEquals(link, expected_link)
```